core/src/main/java/org/apache/iceberg/avro/TypeToSchema.java

```diff
@@ -214,4 +216,58 @@ public Schema primitive(Type.PrimitiveType primitive) {
 
     return primitiveSchema;
   }
+
+  private static String sanitize(String name) {
+    if (validAvroName(name)) {
+      return name;
+    }
+    int length = name.length();
+    StringBuilder sb = new StringBuilder(name.length());
+    char first = name.charAt(0);
+    if (!(Character.isLetter(first) || first == '_')) {
+      sb.append(sanitizedValue(first));
+    } else {
+      sb.append(first);
+    }
+    for (int i = 1; i < length; i++) {
+      char character = name.charAt(i);
+      if (!(Character.isLetterOrDigit(character) || character == '_')) {
```

Comment:
`isLetterOrDigit` corresponds to Avro's rule?

Reply:
Yes, I copied this over from org.apache.avro.Schema#validateName